R/nav-update.R

```diff
@@ -127,3 +127,60 @@ nav_hide <- function(id, target,
                      session = getDefaultReactiveDomain()) {
   shiny::hideTab(id, target, session)
 }
+
+#' Append or prepend nav item to a dropdown menu
+#'
+#' Exported for use by [shiny::prependTab()]/[shiny::appendTab()]. These
+#' functions have been superseded by [nav_insert()] (i.e.,
+#' [shiny::insertTab()]), since it can do everything these functions do (i.e.,
+#' add a [nav()] to the start or end of a [nav_menu()]) and more (i.e., insert a
+#' [nav()] anywhere inside a nav container).
+#'
+#' @inheritParams nav_insert
+#' @param menu_title The title of a [nav_menu()].
+#' @keywords internal
+nav_prepend <- function(id, nav, menu_title, select = FALSE, session = getDefaultReactiveDomain()) {
+
+  force(select)
+  force(menu_title)
+  inputId <- session$ns(id)
+
+  item <- buildTabItem("id", "tsid", TRUE, divTag = tab,
+                       textFilter = if (is.character(tab)) navbarMenuTextFilter else NULL)
+
+  callback <- function() {
+    session$sendInsertTab(
+      inputId = inputId,
+      liTag = processDeps(item$liTag, session),
+      divTag = processDeps(item$divTag, session),
+      menuName = menu_title,
+      target = NULL,
+      position = "before",
```

Comment:
This has changed from `position = "after"`, but the client-side logic has been changed to match https://github.com/rstudio/shiny/pull/3404/files#diff-e14b706227f72447a13adbd59aea9b648d1ade8db57cc03e6b1b8de2eadb2e1bR1018